python/pyspark/sql/functions.py

```diff
@@ -2121,33 +2127,35 @@ def wrapper(*args):
 
         wrapper.func = self.func
         wrapper.returnType = self.returnType
-        wrapper.vectorized = self.vectorized
+        wrapper.pythonUdfType = self.pythonUdfType
 
         return wrapper
 
 
-def _create_udf(f, returnType, vectorized):
+def _create_udf(f, returnType, pythonUdfType):
 
-    def _udf(f, returnType=StringType(), vectorized=vectorized):
-        if vectorized:
+    def _udf(f, returnType=StringType(), pythonUdfType=pythonUdfType):
+        if pythonUdfType == PythonUdfType.PANDAS_UDF \
+           or pythonUdfType == PythonUdfType.PANDAS_GROUPED_UDF:
             import inspect
             argspec = inspect.getargspec(f)
             if len(argspec.args) == 0 and argspec.varargs is None:
                 raise ValueError(
                     "0-arg pandas_udfs are not supported. "
                     "Instead, create a 1-arg pandas_udf and ignore the arg in your function."
```

Comment:
Maybe also update this error message.

Reply:
Thanks! I'll update the message.

Reply:
Hmm, when pandas_grouped_udfs, the number of args should be only 1?

Reply:
I think so. If it didn't become too complicated, maybe we can also check it for pandas_grouped_udf.

Reply:
Thanks, let me try.